### ISSUE TITLE
Editor crashes after editing asset name in Asset Browser

### ISSUE THREAD

**Issue by LB-JaroslawGaweda** [o3de/o3de#12840]
**Describe the bug**
Exiting the edit name mode in the Asset Browser crashes the Editor. Issue occurs on Linux and Windows Github builds. Adding crash logs from windows. 

**Steps to reproduce**
1. Search for any asset in the Asset Browser.
2. Select it and press F2 (or double click on the assets name) to enter name edit mode.
3. Click anywhere else.

**Expected behavior**
You exit the name edit mode.

**Actual behavior**
The Editor crashes.

**Call Stack Trace**
```
Logged at Thu Oct 27 11:55:33 2022
FileVersion: 0.0.0.0
ProductVersion: 0.0.0.0
LogFile: @log@/Editor.log
ProjectDir: C:\Users\GPC-464\O3DE\Projects\NewProject
Executable: Editor.exe (debug: no
Exception Code: 0xC0000005
Exception Addr: 0x0033:0x00007FF85993D73C
Exception Module: 
Exception Description: EXCEPTION_ACCESS_VIOLATION, Attempt to read from address 0x18446744073709551615

The memory could not be "read"

Call Stack Trace:
25) 00007FF8C71F02E7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FF8C6CDD6D8 (InProcessClient64) : (function-name not available)
23) 00007FF8C6CDD321 (InProcessClient64) : (function-name not available)
22) 00007FF8C9A75530 (ntdll) : memset
21) 00007FF8C9A5C876 (ntdll) : _C_specific_handler
20) 00007FF8C9A7241F (ntdll) : _chkstk
19) 00007FF8C9A214A4 (ntdll) : RtlRaiseException
18) 00007FF8C9A70F4E (ntdll) : KiUserExceptionDispatcher
17) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\AssetBrowser\Views\AssetBrowserTreeView.cpp (145) : AzToolsFramework::AssetBrowser::AssetBrowserTreeView::GetSelectedAssets
16) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\AssetBrowser\Views\AssetBrowserTreeView.cpp (674) : AzToolsFramework::AssetBrowser::AssetBrowserTreeView::AfterRename
15) C:\o3de\o3de-packages\packages\qt-5.15.2-rev7-windows\qt\include\QtCore\qobjectdefs_impl.h (443) : QtPrivate::QFunctorSlotObject<`AzToolsFramework::AssetBrowser::EntryDelegate::createEditor'::`2'::<lambda_1>,0,QtPrivate::List<>,void>::impl
14) 00007FF86FC409AA (Qt5Core) : QObject::qt_static_metacall
13) 00007FF87015B078 (Qt5Widgets) : QLineEdit::focusOutEvent
12) 00007FF870067CA1 (Qt5Widgets) : QWidget::event
11) 00007FF87015AB5F (Qt5Widgets) : QLineEdit::event
10) 00007FF8700449A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
 9) 00007FF870043A23 (Qt5Widgets) : QApplication::notify
 8) 00007FF86FC217CB (Qt5Core) : QCoreApplication::notifyInternal2
 7) 00007FF870046583 (Qt5Widgets) : QApplication::setActiveWindow
 6) 00007FF870044342 (Qt5Widgets) : QApplicationPrivate::notifyActiveWindowChange
 5) 00007FF86F40324D (Qt5Gui) : QGuiApplicationPrivate::processActivatedEvent
 4) 00007FF86F3EEE99 (Qt5Gui) : QWindowSystemInterface::sendWindowSystemEvents
 3) 00007FF86FC6A84A (Qt5Core) : QEventDispatcherWin32::processEvents
 2) 00007FF898463799 (qwindows) : qt_plugin_query_metadata
 1) 00007FF86FC1DC5F (Qt5Core) : QEventLoop::exec
```

**Screenshots/Video**

https://user-images.githubusercontent.com/99716227/198257868-3ac45185-fd8e-4240-bf3f-c6f8a07baf44.mp4

**Found in Branch**
Development (8a9db6c)

• Device: PC
• OS: Ubuntu
• Version: Ubuntu 20.04.4 LTS
• CPU Intel Core i7-9700K
• GPU Nvidia GeForce RTX 2070
• Memory 16GB

**Additional context**
[error.log](https://github.com/o3de/o3de/files/9878303/error.log)
[Editor.log](https://github.com/o3de/o3de/files/9878304/Editor.log)
[dump.zip.003.zip](https://github.com/o3de/o3de/files/9878306/dump.zip.003.zip)
[dump.zip.002.zip](https://github.com/o3de/o3de/files/9878307/dump.zip.002.zip)
[dump.zip.001.zip](https://github.com/o3de/o3de/files/9878308/dump.zip.001.zip)

**Comment by amzn-tmryan** [comment]
Looking into this!

**Comment by amzn-tmryan** [comment]
Fixed with PR #12811. Note that you will not be able to rename assets from the table view as mentioned in that PR. We're investigating that issue separately.

**Comment by LB-JakubSkorupka** [comment]
Issue verified as fixed in branch Development (e0e4557) on Windows OS.
